### ISSUE TITLE
Does not support connecting to server that does not use tls-auth

### ISSUE THREAD

**Issue by whohoho** [roburio/openvpn#36]
Its not possible to connect to a server that does not use tls-auth

**Comment by hannesm** [comment]
this is correct. if you're looking for static pre-shared keys mode, please take a look at #11 (which I'm working on, re-started my efforts yesterday evening).

**Comment by hannesm** [comment]
or is there some other authentication mode you'd like to have support for? if it is static pre-shared keys (`secret` in openvpn configuration), I think this is a duplicate of #11

**Comment by whohoho** [comment]
no, i am talking about the optional HMAC signature.

When a server does not have that enabled, and it is enabled on the client, the client cannot connect.

https://openvpn.net/community-resources/hardening-openvpn-security/
The tls-auth HMAC signature provides an additional level of security above and beyond that provided by SSL/TLS. It can protect against: DoS attacks or port flooding on the OpenVPN UDP port. Port scanning to determine which server UDP ports are in a listening state.


packet format here: https://build.openvpn.net/doxygen/network_protocol.html

**Comment by hannesm** [comment]
ok, and do you actively use such a configuration?

**Comment by whohoho** [comment]
I don't actively use such a configuration but its quite common to have openvpn configured that way:

Here is 1 example, checking some other ones to see how they are configured:
https://riseup.net/vpn/vpn-red/windows/windows-riseup.ovpn

**Comment by whohoho** [comment]
another example: https://github.com/leapcode/bitmask-dev/blob/master/src/leap/bitmask/vpn/helpers/linux/bitmask-root#L133

**Comment by cfcs** [comment]
I also think there's value in skipping the computational + bandwidth overhead for an essentially useless HMAC with a global key shared amongst all users. This used to be useful for two things:
- If you trusted all clients, this could be used to limit exposure to `openssl`
- If you had clients behind very simplistic DPI boxes this would make the traffic look less like TLS, but TBH I am not convinced that such boxes exist (they could just check for the x509 certs).

With no `openssl`, and no examples of DPI boxes that are actually fooled by this, I'd be in favor of not having this be mandatory.

**Comment by hannesm** [comment]
see #291, thanks for the report. sorry it took us so long to implement.
